### ISSUE TITLE
feature request: support Cache Directory Tagging Specification

### ISSUE THREAD

**Issue by calestyo** [WayneD/rsync#499]
Hey.

This had already been [asked previously on the mailing list](https://lists.samba.org/archive/rsync/2013-May/028404.html) and a similar solution (`-F`) had been presented there.

However, I think it would still be nice if `rsync` could support the [Cache Directory Tagging Specification](https://bford.info/cachedir/), which basically boil down to, any directory that contains the file `CACHEDIR.TAG` with the content `Signature: 8a477f597d28d172789f06886806bc55` (and optionally some more, see the specs) is not backuped (or on the cased of `rsync` considered for copying).

Apart from avoiding possible side effects of `-F` (which AFAIU has the side effects of `--trust-sender`), the main benefit would IMO be interoperability.
Other tools (e.g. `dar` via its `--cache-directory-tagging` option) already support `CACHEDIR.TAG`, so it would be just nice to be able to use it with `rsync` to, without having to manually "re-build" it with `-F`.

Thanks for consideration,
Chris

**Comment by jkt628** [comment]
it would be nice to have `--exclude-caches` and variants from GNU tar, which were added there back in 2007.

**Comment by idallen** [comment]
Yes, `rsync` would benefit from obeying the `CACHEDIR.TAG` that other programs obey, such as GNU `tar`.  Saying that you can simulate the same thing by adding `rsync`-specific filter files everywhere isn't helpful since that (1) doubles the number of files one has to add everywhere, (2) opens the security risk of `--trust-sender`, and (3) only works for `rsync`.
